authorization-service/handlers/basicAuthorizer.ts

```diff
@@ -0,0 +1,47 @@
+import "source-map-support/register";
+
+export const basicAuthorizer = async (event, _context, callback) => {
+  console.log("Event: ", JSON.stringify(event));
+
+  const generatePolicy = (principalId, resource, effect = "Allow") => {
+    return {
+      principalId: principalId,
+      policyDocument: {
+        Version: "2012-10-17",
+        Statement: [
+          {
+            Action: "execute-api:Invoke",
+            Effect: effect,
+            Resource: resource,
+          },
+        ],
+      },
+    };
+  };
+
+  if (event["type"] !== "TOKEN") callback("Unauthorized");
+
+  try {
+    const authToken = event.authorizationToken;
+    const encodedCreds = authToken.split(" ")[1];
+    const buff = Buffer.from(encodedCreds, "base64");
+    const plainCreds = buff.toString("utf-8").split(":");
+    const username = plainCreds[0];
+    const password = plainCreds[1];
+
+    console.log(`username: ${username} and password ${password}`);
+
+    const storedUserPassword = process.env[username];
+    console.log(storedUserPassword);
+    const effect =
+      !storedUserPassword || storedUserPassword != password ? "Deny" : "Allow";
+
+    const policy = generatePolicy(encodedCreds, event.nethodArn, effect);
```

Comment:
A typo: `nethodArn`. This seems to be the reason why all I see is 403.

Reply:
Thanks! Now it works